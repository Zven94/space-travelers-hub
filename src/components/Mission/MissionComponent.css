@media (prefers-reduced-motion: no-preference) {
  .nav-logo {
    animation: nav-logo-spin infinite 20s linear;
  }
}

.nav-header {
  background-color: #282c34;
  min-height: 75px;
  display: flex;
  flex-direction: row;
  align-items: center;
  justify-content: center;
  font-size: calc(10px + 2vmin);
  color: white;
}

.nav-link {
  color: #61dafb;
}

@keyframes nav-logo-spin {
  from {
    transform: rotate(0deg);
  }

  to {
    transform: rotate(360deg);
  }
}

.nav-logoPlanet {
  width: 50px;
}

.div-table {
  width: 100%;
}
