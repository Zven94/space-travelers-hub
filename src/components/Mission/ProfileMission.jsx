import { useSelector } from 'react-redux';

export default function MissionsList() {
  const { missions } = useSelector((store) => store.missions);
  const missionsList = missions.filter((mission) => mission.status);

  return (
    <>
      <div>
        <h2>My Missions</h2>
        <div>
          {
            missionsList.lenght === 0 ? (
              <h3>No missions yet</h3>
            ) : (
              <ul className="m-0 p-0">
                {missionsList.map((mission, index) => (
                  <li key={mission.id} className={`list-group-item border ${index % 2 === 0 ? 'bg-primary-subtle' : 'bg-white'}`}>
                    {mission.name}
                  </li>
                ))}
              </ul>
            )
          }
        </div>
      </div>
      <div>
        <h2>My Rockets</h2>
                    {reservedRocket.length ? (
              reservedRocket.map((rocket) => (
                <li key={rocket.id} className="list-group-item">
                  {rocket.name}
                </li>
              ))
            ) : (
              <li className="list-group-item">No Reserved Rockets</li>
            )}
      </div>
    </>
  );
}
