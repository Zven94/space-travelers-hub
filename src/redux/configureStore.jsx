import { configureStore } from '@reduxjs/toolkit';
import missionsReducers from './Missions/missionSlice';
import rocketsReducer from './rockets/rocketsSlice';

export default configureStore({
  reducer: {
    missions: missionsReducers,
    rockets: rocketsReducer,
  },
});
