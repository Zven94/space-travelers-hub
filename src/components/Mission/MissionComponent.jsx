import './MissionComponent.css';
import Navbar from '../Navbar/Navbar';

function MissionComponent() {

  return (
    <Navbar/>
  );
}

export default MissionComponent;
