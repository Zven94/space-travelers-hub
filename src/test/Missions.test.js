import { Provider } from 'react-redux';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import { BrowserRouter as Router } from 'react-router-dom';
import configureStore from '../redux/configureStore';
import Mission from '../router/Missions';
import MissionComponent from '../components/Mission/MissionComponent';
import MissionsList from '../components/Mission/ProfileMission';

describe('test for missions components', () => {
  const mission = (
    <Provider store={configureStore}>
      <Router>
        <Mission />
      </Router>
    </Provider>
  );

  test('Take snapshot from the element', () => {
    const missionComponent = render(mission);
    expect(missionComponent).toMatchSnapshot();
  });

  const table = (
    <Provider store={configureStore}>
      <Router>
        <MissionComponent />
      </Router>
    </Provider>
  );

  test('Take snapshot from the element', () => {
    const tableComponent = render(table);
    expect(tableComponent).toMatchSnapshot();
  });

  const list = (
    <Provider store={configureStore}>
      <Router>
        <MissionsList />
      </Router>
    </Provider>
  );

  test('Take snapshot from the element', () => {
    const listComponent = render(list);
    expect(listComponent).toMatchSnapshot();
  });
});
