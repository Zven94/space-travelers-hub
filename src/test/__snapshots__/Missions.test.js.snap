// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test for missions components Take snapshot from the element 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <main
        class=""
      >
        <div
          class="nav-header justify-content-around ms-auto nav"
        >
          <div
            class="container"
          >
            <div
              class="row"
            >
              <div
                class="d-flex flex-row aling-item-center text-center col"
              >
                <img
                  alt="logoPlanet"
                  class="nav-logoPlanet"
                  src="logo-planet.png"
                />
                <h2
                  class="m-0 pt-1"
                >
                  Space Traveler\`s Hub
                </h2>
              </div>
              <div
                class="d-flex flex-row aling-item-center justify-content-around col"
              >
                <a
                  aria-current="page"
                  class="text-decoration-none active"
                  href="/"
                >
                   Rockets 
                </a>
                <a
                  class="text-decoration-none"
                  href="/missions"
                >
                   Missions
                </a>
                <a
                  class="text-decoration-none"
                  href="/profile"
                >
                  My Profile
                </a>
              </div>
            </div>
          </div>
        </div>
      </main>
      <div
        class="container"
      >
        <div
          class="justify-content-center align-items-center col"
        >
          <h2>
            Missons
          </h2>
          <table
            class="table table-striped table-bordered"
          >
            <thead>
              <tr>
                <th>
                  Missions
                </th>
                <th>
                  Description
                </th>
                <th>
                  Status
                </th>
                <th>
                  Action
                </th>
              </tr>
            </thead>
            <tbody />
          </table>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <main
      class=""
    >
      <div
        class="nav-header justify-content-around ms-auto nav"
      >
        <div
          class="container"
        >
          <div
            class="row"
          >
            <div
              class="d-flex flex-row aling-item-center text-center col"
            >
              <img
                alt="logoPlanet"
                class="nav-logoPlanet"
                src="logo-planet.png"
              />
              <h2
                class="m-0 pt-1"
              >
                Space Traveler\`s Hub
              </h2>
            </div>
            <div
              class="d-flex flex-row aling-item-center justify-content-around col"
            >
              <a
                aria-current="page"
                class="text-decoration-none active"
                href="/"
              >
                 Rockets 
              </a>
              <a
                class="text-decoration-none"
                href="/missions"
              >
                 Missions
              </a>
              <a
                class="text-decoration-none"
                href="/profile"
              >
                My Profile
              </a>
            </div>
          </div>
        </div>
      </div>
    </main>
    <div
      class="container"
    >
      <div
        class="justify-content-center align-items-center col"
      >
        <h2>
          Missons
        </h2>
        <table
          class="table table-striped table-bordered"
        >
          <thead>
            <tr>
              <th>
                Missions
              </th>
              <th>
                Description
              </th>
              <th>
                Status
              </th>
              <th>
                Action
              </th>
            </tr>
          </thead>
          <tbody />
        </table>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`test for missions components Take snapshot from the element 2`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <main
        class=""
      >
        <div
          class="nav-header justify-content-around ms-auto nav"
        >
          <div
            class="container"
          >
            <div
              class="row"
            >
              <div
                class="d-flex flex-row aling-item-center text-center col"
              >
                <img
                  alt="logoPlanet"
                  class="nav-logoPlanet"
                  src="logo-planet.png"
                />
                <h2
                  class="m-0 pt-1"
                >
                  Space Traveler\`s Hub
                </h2>
              </div>
              <div
                class="d-flex flex-row aling-item-center justify-content-around col"
              >
                <a
                  aria-current="page"
                  class="text-decoration-none active"
                  href="/"
                >
                   Rockets 
                </a>
                <a
                  class="text-decoration-none"
                  href="/missions"
                >
                   Missions
                </a>
                <a
                  class="text-decoration-none"
                  href="/profile"
                >
                  My Profile
                </a>
              </div>
            </div>
          </div>
        </div>
      </main>
      <div
        class="container"
      >
        <div
          class="justify-content-center align-items-center col"
        >
          <h2>
            Missons
          </h2>
          <table
            class="table table-striped table-bordered"
          >
            <thead>
              <tr>
                <th>
                  Missions
                </th>
                <th>
                  Description
                </th>
                <th>
                  Status
                </th>
                <th>
                  Action
                </th>
              </tr>
            </thead>
            <tbody />
          </table>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <main
      class=""
    >
      <div
        class="nav-header justify-content-around ms-auto nav"
      >
        <div
          class="container"
        >
          <div
            class="row"
          >
            <div
              class="d-flex flex-row aling-item-center text-center col"
            >
              <img
                alt="logoPlanet"
                class="nav-logoPlanet"
                src="logo-planet.png"
              />
              <h2
                class="m-0 pt-1"
              >
                Space Traveler\`s Hub
              </h2>
            </div>
            <div
              class="d-flex flex-row aling-item-center justify-content-around col"
            >
              <a
                aria-current="page"
                class="text-decoration-none active"
                href="/"
              >
                 Rockets 
              </a>
              <a
                class="text-decoration-none"
                href="/missions"
              >
                 Missions
              </a>
              <a
                class="text-decoration-none"
                href="/profile"
              >
                My Profile
              </a>
            </div>
          </div>
        </div>
      </div>
    </main>
    <div
      class="container"
    >
      <div
        class="justify-content-center align-items-center col"
      >
        <h2>
          Missons
        </h2>
        <table
          class="table table-striped table-bordered"
        >
          <thead>
            <tr>
              <th>
                Missions
              </th>
              <th>
                Description
              </th>
              <th>
                Status
              </th>
              <th>
                Action
              </th>
            </tr>
          </thead>
          <tbody />
        </table>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`test for missions components Take snapshot from the element 3`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <ul
        class="m-0 p-0"
      />
    </div>
  </body>,
  "container": <div>
    <ul
      class="m-0 p-0"
    />
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
